src/react/utils.js

```diff
@@ -264,3 +263,35 @@ export function normalizeFunctionalComponentParamaters(func: ECMAScriptSourceFun
     }
   });
 }
+
+export function getThisAssignments(functionBody: BabelNodeBlockStatement): Set<string> {
+  let assignments = new Set();
+  traverse(
+    t.file(t.program([t.functionDeclaration(t.identifier("dummy"), [], functionBody)])),
+    {
+      "ThisExpression|Identifier": function(path) {
+        if (t.isThisExpression(path.node) || path.node.name === "this") {
```

Comment:
Is there any time that a ThisExpression doesn't have `name === "this"`?

Reply:
`ThisExpression` doesn't have a `name`. This checks if the `Identifier` has a `name` of `this` (it can happen now and then, so this checks against it).